library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java

```diff
@@ -84,18 +84,15 @@ long getBatchSizeInBytes(long batchId) {
         long totalSize = 0;
         try {
             String[] selectionArgs = {String.valueOf(batchId)};
-            cursor = resolver.query(ALL_DOWNLOADS_CONTENT_URI,
-                    null,
+            cursor = resolver.query(
+                    ALL_DOWNLOADS_CONTENT_URI,
+                    new String[]{"sum(" + COLUMN_TOTAL_BYTES + ")"},
                     COLUMN_BATCH_ID + " = ?",
                     selectionArgs,
                     null);
 
-            int totalBytesColumnIndex = cursor.getColumnIndexOrThrow(COLUMN_TOTAL_BYTES);
-
-            while (cursor.moveToNext()) {
-                int individualDownloadSize = cursor.getInt(totalBytesColumnIndex);
-                totalSize += individualDownloadSize;
-            }
+            cursor.moveToFirst();
+            totalSize = cursor.getLong(0);
```

Comment:
what is this column id? I thought you needed to define another column variable in the raw query and then read from that?